GetInteriorVehicleDataCapabilities API of Reverse SDL

dev-gh opened the pull request:
Changes contain 
- working RSDL build
- basic implementation of GetInteriorVehicleDataCapabilities RPC
- some RPCs have been disabled according to customer proposal (https://github.com/yang1070/sdl_evolution/blob/170b85f894fc64f763b6c600708ff5683004ac89/proposals/NNNN-remote-control-baseline.md)
- code style changes with clang
- bugfixes for REMOTE_CONTROL enum conversion, RC RPCs permissions validation, GetInteriorVehicleDataCapabilities response validation

AByzhynar committed:
Add possibility to build SDL with RC functionality

Added necessary flags, variables and their handling to make possible to
build SDL with RC functionality

AByzhynar committed:
Update config profle to support plugin folder

Added handling of plugin folder.
Changed one of profile interfaces

Fix UT

AByzhynar committed:
Update connection handler OnDeviceRemove() method

Moved device erasing from devices list after connection observer device
removal

AByzhynar committed:
Remove redundant code from formatters & smart objects unit tests

Removed commented code

AByzhynar committed:
Update security manager

Added mandatory header files includes to SSL context
Added funcion GetName to get certificate name and some logging to SSL
context

AByzhynar committed:
Update media manager and related unit tests

Updated header guards
Included necessary paths, added libriries and added
RC fuctionality support to CMakeLists
Added REMOTE_CONTROL handling to CmakeLists
Replaced raw ptrs with SharedPtr in setters for unit tests
Added/updated unit tests for media manager

AByzhynar committed:
Add rc functionality support & fix memory leaks

Added TimeTester library creation and included RC functionality
directories
Fixed some memory leaks in utils & telemetry monitor
Updated related unit tests

AByzhynar committed:
Update utils component

Added export function definition
Added case sensitive comparison methods
Update UT macro FRIEND_TEST
Added timer_thread class
Updated utils unit tests

AKalinich-Luxoft committed:
interfaces applicaiton mock

LuxoftAKutsan committed:
Functional module implementation

Commit:
Remove unused ifndef __QNXNTO__

LuxoftAKutsan committed:
Policy and AM additional files for RevSDL

Commit:
Adds RC-related changes for appMain folder

Commit:
Adds RC-related changes to HMI message handler folder

Commit:
Removes unused subscription, improves style, adds base classes virtual dtors

AByzhynar committed:
RC support by external policy part1

Commit:
Full copy of CAN cooperation

AKalinich-Luxoft committed:
Moved RC functions for AM header files

Commit:
Removes zones concept from RC implementation

Commit:
Removes RPCs implemenations

dev-gh committed:
Merge pull request #117 from dev-gh/feature/rc_baseline_appmain_hmi_handler

Adds RC-related parts to appMain and HMI message handler

AKalinich-Luxoft committed:
Moved rc-related functions implementation for AM interface

AKalinich-Luxoft committed:
Moved RC-related mock functions for AM tests

AKalinich-Luxoft committed:
Fixed formatting with clang

Commit:
Removes zones from JSON sources

LuxoftAKutsan committed:
Fix build issues with policies and Application manager

Commit:
Removes odd parts from CMake file

LuxoftAKutsan committed:
Merge pull request #120 from LuxoftAKutsan/feature_part/can_cooperation_and_plugin_manager

feature part/can cooperation and plugin manager

LuxoftAKutsan committed:
Merge pull request #116 from LitvinenkoIra/feature/4.4.0_rc_functionality

Move some changes related to RC

LuxoftAKutsan committed:
Merge branch 'feature/sdl_remote_control_baseline' into feature/rc_baseline_can_cooperation

LuxoftAKutsan committed:
Merge pull request #118 from dev-gh/feature/rc_baseline_can_cooperation

Adds CAN cooperation part of RC implementation

LuxoftAKutsan committed:
Merge pull request #119 from AByzhynar/feature/Policy_RC

RC support by external policy part1

LuxoftAKutsan committed:
Fix collecting sources in can_cooperation

LuxoftAKutsan committed:
Remove CheckAccess function from policies because this function was used only with zones

LuxoftAKutsan committed:
Merge pull request #121 from AKalinich-Luxoft/move_rc_functionality_application_manager

Move RC functionality for Application Manager and related classes

LuxoftAKutsan committed:
Remove redundant plugins folder

LuxoftAKutsan committed:
Remove merge artifacts

LuxoftAKutsan committed:
Hmi message handler

LuxoftAKutsan committed:
AM build fixes

AByzhynar committed:
App manager & policy handler fixes for RC

AKalinich-Luxoft committed:
Added missed #endif

AByzhynar committed:
Add missed #ifdef RC to mock_message_helper

Added missed #ifdef wrapper for several message_helper methods related to RC
functionality

LuxoftAKutsan committed:
Remove merge issues from applicaion mock

Commit:
RC support by regular policy

LuxoftAKutsan committed:
AM and policies and can module build fixes

LuxoftAKutsan committed:
UT RC ON EXTERNAL build :-)

LuxoftAKutsan committed:
Merge pull request #122 from LitvinenkoIra/feature/rc_regular_policy

RC support by regular policy

Commit:
Extends structs and adds GetInteriorVehicleCapabilities request/response

New Reverse SDL API GetInteriorVehicleCapabilities is introduced with
appropriate extentions for existing API values.
Also did cleanup of trailing and odd whitespaces

Commit:
Cleans up HMI_API from odd/trailing whitespaces

Commit:
Adds GetInteriorVehicleDataCapabilities with structs to HMI API

New Reverse SDL API is added according to MOBILE API changes.
Also updates also ButtonPress params due to removed module zones logic and
parameters.

Commit:
Adds draft GetInteriorVehicleDataCapabilities implemenation and validation

Basic implementation of request/response handling in plugin logic.

Commit:
Style changes, formats sources with clang-format

Commit:
Enables SDL processing of GetInteriorVehicleDataCapabilities request/response

Also some APIs have been disabled due to customer proposal

Commit:
Changes order of HMI handler setup for proper plugin subsystem initialization

In order to start properly plugin subsystem requires HMI handler to be in place
during initialization, so changed the order to have handler setup before.

Commit:
Removes modules zones validation due to customer proposal requirement

By proposal zones must be excluded from API and Reverse SDL logic.

Commit:
Fix sql queries

LuxoftAKutsan committed:
Merge pull request #123 from LitvinenkoIra/fix/sql_queries

Fix sql queries

Commit:
Makes registered Reverse SDL application device be always assigned as driver's

By customer proposal all devices having Reverse SDL applications must be
considered as driver's and be processed accordingly.

Commit:
Disables validation of HMI response for GetInteriorVehicleDataCapabilities

There was incorrect validator used so just replaced its usage with plain
copying of HMI response.

Commit:
Stub for GetInteriorVehicleDataCapabilites response validation

Bare skeleton classes to be filled with proper logic later.

Commit:
Fix SQL PT Representation tests failure

LuxoftAKutsan committed:
Merge pull request #125 from LitvinenkoIra/fix/policy_test_failure

Fix SQL PT Representation tests failure

Commit:
Fixes RPCs permissions check for enabled Reverse SDL applications

RSDL applications must consider only 'groups_primaryRC' and
'groups_nonprimaryRC' but not the usual 'groups'

AByzhynar commented on src/components/can_cooperation/include/can_cooperation/commands/get_interior_vehicle_data_capabilities_request.h:
```diff
@@ -0,0 +1,103 @@
+  void UpdateModules(Json::Value* params);
+  application_manager::TypeAccess CheckModuleTypes(const Json::Value& message);
+  bool ReadCapabilitiesFromFile();
+  void CreateCapabilities(const Json::Value& zone_capabilities,
```
@dev-gh `zone_capabilities`? Are you sure that we have to handle this parameter?

dev-gh replied:
@AByzhynar 
Currently there is no clarification from customer on format expected for fallback capabilities, so I would leave these parts untill further information will be provided and than change that accordingly.

AByzhynar commented on src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc:
```diff
@@ -0,0 +1,311 @@
+  }
+
+  VehicleCapabilities file_caps;
+  Json::Value zone_capabilities;
```
@dev-gh The same question as above: `Zone_capabilities`? Do we need it?

dev-gh replied:
@AByzhynar 
Currently there is no clarification from customer on format expected for fallback capabilities, so I would leave these parts untill further information will be provided and than change that accordingly.

AByzhynar commented on src/components/can_cooperation/src/validators/get_interior_vehicle_data_capabilities_response_validator.cc:
```diff
@@ -0,0 +1,64 @@
+
+}  // namespace valdiators
+
+}  // namespace can_cooperation
```
@dev-gh Please add one empty line at the end of all files

dev-gh replied:
@AByzhynar 
Done with 40ce3c6

Commit:
Fixes REMOTE_CONTROL enum value conversion for policy database

Commit:
Fixes verification of HMI response for GetInteriorVehicleDataCapabilities

According to new API response must contain not array but
RemoteControlCapabilities struct.

dev-gh commented:
@LuxoftAKutsan @LitvinenkoIra 
Please review

LuxoftAKutsan commented on src/components/application_manager/include/application_manager/message.h:
```diff
@@ -92,7 +92,8 @@ class Message {
 
-  //! --------------------------------------------------------------------------.
+  //!
+  //--------------------------------------------------------------------------.
```
@dev-gh that does this changes mean?

LuxoftAKutsan commented on src/components/application_manager/test/include/application_manager/mock_application.h:
```diff
@@ -285,8 +285,9 @@ class MockApplication : public ::application_manager::Application {
 #ifdef SDL_REMOTE_CONTROL
-  MOCK_METHOD1(set_system_context,
-               void(const application_manager::mobile_api::SystemContext::eType&));
+  MOCK_METHOD1(
```
@dev-gh please avoid style check in your PR

LuxoftAKutsan commented on src/components/can_cooperation/src/can_module.cc:
```diff
@@ -226,76 +226,83 @@ functional_modules::ProcessResult CANModule::HandleMessage(
-      }
-      return ProcessResult::CANNOT_PROCESS;
-    }
+    // Disabled
```
@dev-gh please make it as TODO with description why it is disabled

Commit:
Adds newlines at the end of files

Commit:
Fixes application activation flow

Enabled processing of SDL.ActivateApp via plugin system.

Commit:
Fixes setting of default HMI level for non-EXTENDED policy

LuxoftAKutsan reviewed (commented):
@dev-gh changes rebased and cherry picked
